backend/de.metas.manufacturing/src/main/java/de/metas/manufacturing/generatedcomponents/ManufacturingComponentGeneratorService.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * #%L
+ * de.metas.manufacturing
+ * %%
+ * Copyright (C) 2020 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.manufacturing.generatedcomponents;
+
+import de.metas.javaclasses.IJavaClassBL;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.exceptions.AdempiereException;
+import org.adempiere.mm.attributes.api.ImmutableAttributeSet;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ManufacturingComponentGeneratorService
+{
+	private final ComponentGeneratorRepository componentRepository;
+	private final IJavaClassBL javaClassBL = Services.get(IJavaClassBL.class);
+
+	// TODO tbp: add CCache for productId->PP_ComponentGenerator.
+
+	public ManufacturingComponentGeneratorService(
+			@NonNull final ComponentGeneratorRepository componentRepository
+	)
+	{
+		this.componentRepository = componentRepository;
+	}
+
+	public ImmutableAttributeSet generate(@NonNull final GeneratedComponentRequest request)
+	{
+		final PP_ComponentGenerator generator = componentRepository.getByProductId(request.getProductId());
```

Comment:
avoid throwing exception because the caller has no way to know which product has a generator and which one does not.
return ImmutableAttributeSet.EMPTY

Reply:
Is it really better to silently fail (create no data), than to inform the user that his (expected) config is not set?

Reply:
solved by adding a new method `hasGeneratorForProduct`